Sources/Confidence/Confidence.swift

```diff
@@ -1,23 +1,40 @@
 import Foundation
 
 public class Confidence: ConfidenceEventSender {
-    public var context: ConfidenceStruct
+    private let parent: ConfidenceEventSender?
+    private var localContext: ConfidenceStruct
+
+    public var context: ConfidenceStruct {
+        get {
+            var mergedContext = parent?.context ?? [:]
+            localContext.forEach { entry in
+                mergedContext.updateValue(entry.value, forKey: entry.key)
```

Comment:
here we should also filter the keys that are removed from this instance. if the parent has a key named `my_key` and we remove the `my_key` context from the child instance. when we want to calculate the context we should filter that key out.